MyModels.py

```diff
@@ -0,0 +1,158 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
```

Comment:
This index is not updated by the BaseModel itself. Let's move that logic inside, as it's the most straightforward compared to other indexes.

Reply:
You are right! this situation has been created after editing code and simplifying the basemodel